packages/data/src/world/oot/overworld.yml

```diff
@@ -2080,12 +2080,28 @@
     "Lake Hylia Gossip Near River": "true"
     "Lake Hylia Gossip Southeast Lake": "true"
     "Lake Hylia Gossip Southwest Lake": "true"
+"Lake Hylia Near Shortcut":
+  region: LAKE_HYLIA
+  age_change: false
+  exits:
+    "Lake Hylia": "is_adult || can_swim"
+    "Zora Domain Shortcut": "is_adult || can_swim"
+"Lake Hylia Near Pond":
+  region: LAKE_HYLIA
+  time: flow
+  exits:
+    "Fishing Pond": "true"
+    "Lake Hylia": "is_adult || can_swim_or_sink"
```

Comment:
This could potentially be problematic for adult if the default state of lake hylia isn't just always empty even after finishing Water Temple, because what if this is your first entry into Lake Hylia. I know we were going to handle it from the Domain entrance, but this one would also be an issue.